### PR TITLE
Feature: Royalty fee and local marketplace listing fees update

### PR DESCRIPTION
**Pull request by chexware**

### CODE REVIEW

File: pallets/auction/src/lib.rs [hunk 1]
```diff
@@ -956,22 +958,20 @@ pub mod pallet {
...
-			// asset_id.1)).ok_or(Error::<T>::AssetIsNotExist)?;
-
-			let class_fund = T::NFTHandler::get_class_fund(&asset_id.0);
+			let nft_details = T::NFTHandler::get_nft_detail((asset_id.0, asset_id.1))?;
+			let mut royalty_fee = nft_details.royalty_fee * *high_bid_price;
+			let mut class_fund = T::NFTHandler::get_class_fund(&asset_id.0);
+			match listing_level {
+				ListingLevel::Local(metaverse_id) => {
+					class_fund = T::MetaverseInfoSource::get_metaverse_treasury(metaverse_id);
+					royalty_fee =
```
Comment (justinphamnz) [inline pallets/auction/src/lib.rs:970]: royalty_fee and listing fee are 2 different concepts. Royalty fee will go to NFT fund, listing fee will go to metaverse treasury.

Comment (chexware) [reply pallets/auction/src/lib.rs:970]: sure will update the code to support both

**Review by justinphamnz — CHANGES_REQUESTED** [review]
Good work @chexware. Listing fee info should be included inside Auction struct when new auction created. It will prevent listing fee changes after auction started

File: pallets/auction/src/lib.rs (hunk 1 above)
Comment (chexware) [reply pallets/auction/src/lib.rs:970]: @justinphamnz  do the global marketplace have a listing fee as well?

**Comment by chexware** [comment]
@justinphamnz  I've updated the auction items to have fee collection property so the metaverse owner can't change their listing fee during the auction. Also, when an auction is finalized booth royalty fee and listing fee are collected. Let me know if I need to change anything else

**Comment by chexware** [comment]
@justinphamnz I've updated the auction item type to include listing fee property so metaverse owner can't change listing fee after an auction/buy_now is created. Also updated the code so both listing fee and royalty fees are collected when the auction is finalized. Let me know if I need to make any other changes.

**Pull request merged**